### PR TITLE
🎮 Add urn to step/message context and make urn scheme accessible for an…

### PR DESCRIPTION
**Pull request by norkans7**
…on org

### CODE REVIEW

File: temba/msgs/models.py [hunk 1]
```diff
@@ -1404,14 +1404,19 @@ def build_expressions_context(self):
...
+        context = {
             "__default__": value,
             "value": value,
             "text": self.text,
             "attachments": attachments,
             "time": datetime_to_str(self.created_on, format=date_format, tz=self.org.timezone),
         }
 
+        if self.contact_urn:
+            context["urn"] = self.contact.get_urn_context(self.org, scheme=self.contact_urn.scheme)
```
Comment (rowanseymour) [inline temba/msgs/models.py]: isn't this potentially the wrong URN? it should be `self.contact_urn`

Comment (norkans7) [reply temba/msgs/models.py]: You mean if we have 2 urns withe same scheme?

Comment (rowanseymour) [reply temba/msgs/models.py]: yeah you have the URN already as `self.contact_urn` so seems odd to go and look it up again on the contact and hope it's correct

File: temba/contacts/models.py [hunk 2]
```diff
@@ -2538,6 +2533,18 @@ def lookup(cls, org, urn_as_string, country_code=None, normalize=True):
 
         return existing
 
+    def build_expressions_context(self, org):
```
Comment (norkans7) [inline temba/contacts/models.py]: We need to pass the org since urns are cached

File: temba/flows/models.py [hunk 3]
```diff
@@ -1693,6 +1693,7 @@ def build_expressions_context(self, contact, msg, run=None):
 
             # only populate channel if this contact can actually be reached (ie, has a URN)
             if contact_urn:
+                message_context["urn"] = contact_urn.build_expressions_context(self.org)
```
Comment (rowanseymour) [inline temba/flows/models.py]: isn't this set in `Msg.build_expressions_context()` called on line 1680 ?

Comment (norkans7) [reply temba/flows/models.py]: Yes, I think the resolve_recipient might have urn that is going to be used below and `Msg.build_expressions_context()` only add urn if we have one set on the message

Comment (rowanseymour) [reply temba/flows/models.py]: `@step.urn` (`@input.urn` in new engine) should be the URN on the message, no? I don't get why we're overwriting it here

Comment (norkans7) [reply temba/flows/models.py]: I believe this is the case where we do not yet have a preferred channel for a contact and we are trying to find which channel to use

Comment (rowanseymour) [reply temba/flows/models.py]: sure sure I get that for `@channel`, but for `@input.urn` we want that to be the URN of the message - not a hypothetical URN.

Comment (norkans7) [reply temba/flows/models.py]: OK

**Pull request merged**